### PR TITLE
Add image cta in peer profile

### PR DESCRIPTION
**Pull request by penk85**
WPC-434: Add an image Call To Action (image and URL) inside the peer profile.

### CODE REVIEW

**Review by bishoymelek — APPROVED** [review]
:tada:

**Review by HanaAyman — APPROVED** [review]

File: packages/api/__tests__/specs/__snapshots__/peer.ts.snap [hunk 1]
```diff
@@ -135,6 +137,8 @@ exports[`Peers can be created/edited/deleted: can update peer profile 1`] = `
 Object {
   "data": Object {
     "updatePeerProfile": Object {
+      "callToActionImage": null,
+      "callToActionImageURL": null,
```
Comment (roosnic1) [inline packages/api/__tests__/specs/__snapshots__/peer.ts.snap:141]: here it's `null` and in the test snapshot above it's `""`. I think it should be the same in both.

File: packages/editor/src/client/panel/peerProfileEditPanel.tsx [hunk 2]
```diff
@@ -33,7 +33,10 @@ export function PeerInfoEditPanel({onClose, onSave}: ImageEditPanelProps) {
   const [name, setName] = useState('')
   const [themeColor, setThemeColor] = useState('')
   const [callToActionText, setCallToActionText] = useState<RichTextBlockValue>(createDefaultValue())
-  const [callToActionURL, setCallToActionURL] = useState('')
+  const [callToActionTextURL, setCallToActionTextURL] = useState('')
+  const [callToActionImageID, setCallToActionImageID] = useState<PeerProfileImage>()
```
Comment (roosnic1) [inline packages/editor/src/client/panel/peerProfileEditPanel.tsx:37]: I'm not sure if the `PeerProfileImage` is the right type. It's used for the peer profile and you have here just a normal image.

File: packages/editor/src/client/panel/peerProfileEditPanel.tsx [hunk 3]
```diff
@@ -144,15 +204,22 @@ export function PeerInfoEditPanel({onClose, onSave}: ImageEditPanelProps) {
...
+            isLogoChange ? setLogoImage(value) : setCallToActionImageID(value)
           }}
         />
       </Drawer>
 
       <Drawer show={isEditModalOpen} size={'sm'} onHide={() => setEditModalOpen(false)}>
-        {logoImage && (
-          <ImagedEditPanel id={logoImage!.id} onClose={() => setEditModalOpen(false)} />
-        )}
+        {isLogoChange
```
Comment (roosnic1) [inline packages/editor/src/client/panel/peerProfileEditPanel.tsx:213]: this is a lot of repetitive code i think you can put must of the logic in side the `id={...}` expression.

**Review by roosnic1 — CHANGES_REQUESTED** [review]

**Review by HanaAyman — APPROVED** [review]

File: packages/api-db-mongodb/src/db/peer.ts [hunk 4]
```diff
@@ -30,8 +30,8 @@ export class MongoDBPeerAdapter implements DBPeerAdapter {
         themeColor: '#000000',
         callToActionURL: '',
         callToActionText: [],
-        callToActionImageID: '',
-        callToActionImageURL: ''
+        callToActionImageID: undefined,
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/db/peer.ts:33]: they are undefined by default. I think you should set them to `''`

File: packages/editor/src/client/panel/peerProfileEditPanel.tsx [hunk 5]
```diff
@@ -60,7 +62,7 @@ export function PeerInfoEditPanel({onClose, onSave}: ImageEditPanelProps) {
           : createDefaultValue()
       )
       setCallToActionTextURL(data.peerProfile.callToActionURL)
-      setCallToActionImageID(data.peerProfile.callToActionImage)
+      setCallToActionImage(data?.peerProfile?.callToActionImage ?? undefined)
```
Comment (roosnic1) [inline packages/editor/src/client/panel/peerProfileEditPanel.tsx:65]: you don't need the `?? undefined` if you use them before the dots.

**Review by roosnic1 — CHANGES_REQUESTED** [review]

**Review by HanaAyman — APPROVED** [review]

**Review by roosnic1 — DISMISSED** [review]

**Review by bishoymelek — APPROVED** [review]
:tada:

**Review by HanaAyman — APPROVED** [review]

**Review by roosnic1 — APPROVED** [review]
👍🏾

**Pull request merged**